### PR TITLE
[Merged by Bors] - clippy macOS

### PR DESCRIPTION
**Pull request by aviramha**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -68,8 +68,26 @@ jobs:
       - uses: Swatinem/rust-cache@v1
       # Needed for agent
       - run: cargo fmt --all -- --check
+      - run: rustup target add --toolchain nightly aarch64-apple-darwin
```
Comment (t4lz) [inline .github/workflows/ci.yaml:71]: Does this have an affect? `--all-targets` is not needed for that?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -68,8 +68,26 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          components: rustfmt, clippy
+          target: aarch64-apple-darwin
+      - uses: Swatinem/rust-cache@v1
+      # only mirrord, mirrord-sip contain macOS specific, so run clippy on that.
```
Comment (t4lz) [inline .github/workflows/ci.yaml:87]: Would it be notably slower to run it on all the code? That way we wouldn't have to remember to add stuff to here if we add MacOS-specific code in the future.

File: mirrord/cli/src/main.rs [hunk 3]
```diff
@@ -9,8 +9,6 @@ use anyhow::{anyhow, Context, Result};
 use clap::Parser;
 use config::*;
 use const_random::const_random;
-#[cfg(all(target_os = "macos", target_arch = "aarch64"))]
-use errno;
```
Comment (t4lz) [inline mirrord/cli/src/main.rs:13]: How does that work if we use `errno` there: https://github.com/metalbear-co/mirrord/blob/a76adadaca201b4e9408e85df2ad8bea10728987/mirrord/cli/src/main.rs#L234?

**Review by t4lz — APPROVED** [review]

File: mirrord/cli/src/main.rs (hunk 3 above)
Comment (aviramha) [reply mirrord/cli/src/main.rs:13]: `errno:Errno` is "fully qualified" name. In Rust `use errno` usually has no particular meaning

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (aviramha) [reply .github/workflows/ci.yaml:87]: Problem is not all code compiles for macOS (agent)..

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (aviramha) [reply .github/workflows/ci.yaml:71]: oops, my bad.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (aviramha) [reply .github/workflows/ci.yaml:86]: (It can't compile because of dependencies)

**Pull request closed without merging**